packages/sdk/src/evm/contracts/prebuilt-implementations/loyalty-card.ts

```diff
@@ -0,0 +1,390 @@
+import { NFT, NFTMetadataOrUri } from "../../../core/schema/nft";
+import { getRoleHash } from "../../common/role";
+import { buildTransactionFunction } from "../../common/transactions";
+import { ContractAppURI } from "../../core/classes/contract-appuri";
+import { ContractEncoder } from "../../core/classes/contract-encoder";
+import { ContractEvents } from "../../core/classes/contract-events";
+import { ContractInterceptor } from "../../core/classes/contract-interceptor";
+import { ContractMetadata } from "../../core/classes/contract-metadata";
+import { ContractOwner } from "../../core/classes/contract-owner";
+import { ContractPlatformFee } from "../../core/classes/contract-platform-fee";
+import { ContractRoles } from "../../core/classes/contract-roles";
+import { ContractRoyalty } from "../../core/classes/contract-royalty";
+import { ContractPrimarySale } from "../../core/classes/contract-sales";
+import { ContractWrapper } from "../../core/classes/contract-wrapper";
+import { StandardErc721 } from "../../core/classes/erc-721-standard";
+import { Erc721WithQuantitySignatureMintable } from "../../core/classes/erc-721-with-quantity-signature-mintable";
+import { GasCostEstimator } from "../../core/classes/gas-cost-estimator";
+import { Transaction } from "../../core/classes/transactions";
+import type { NetworkInput, TransactionResultWithId } from "../../core/types";
+import { AddressOrEns } from "../../schema/shared/AddressOrEnsSchema";
+import { Address } from "../../schema/shared/Address";
+import { Abi, AbiInput, AbiSchema } from "../../schema/contracts/custom";
+import { LoyaltyCardContractSchema } from "../../schema/contracts/loyalty-card";
+import { SDKOptions } from "../../schema/sdk-options";
+import type { LoyaltyCard as LoyaltyCardContract } from "@thirdweb-dev/contracts-js";
+import { ThirdwebStorage } from "@thirdweb-dev/storage";
+import { BigNumberish, CallOverrides, constants } from "ethers";
+import { NFT_BASE_CONTRACT_ROLES } from "../contractRoles";
+
+/**
+ * Create a collection of one-of-one NFTs.
+ *
+ * @example
+ *
+ * ```javascript
+ * import { ThirdwebSDK } from "@thirdweb-dev/sdk";
+ *
+ * const sdk = new ThirdwebSDK("{{chainName}}");
+ * const contract = await sdk.getContract("{{contract_address}}", "nft-collection");
+ * ```
+ *
+ * @public
+ */
+export class LoyaltyCard extends StandardErc721<LoyaltyCardContract> {
```

Comment:
I don't think we need a prebuilt class for this, we want to move away from specific prebuilt classes in favor of using SmartContract + extension.

The API for loyalty should just be `contract.erc721.loyalty.xyz()`

this makes it more straightforward for users (one way of doing things) and way more maintainable for us.

In a few versions of the SDK the goal is to remove all the prebuilt classes